sklearnex/svm/svc.py

```diff
@@ -41,46 +39,58 @@ def __init__(self, *, C=1.0, kernel='rbf', degree=3, gamma='scale',
             random_state=random_state)
 
     def fit(self, X, y, sample_weight=None):
-        if self.kernel in ['linear', 'rbf', 'poly', 'sigmoid']:
-            logging.info("sklearn.svm.SVC.fit: " + get_patch_message("onedal"))
-            self._onedal_fit(X, y, sample_weight)
-        else:
-            logging.info("sklearn.svm.SVC.fit: " + get_patch_message("sklearn"))
-            sklearn_SVC.fit(self, X, y, sample_weight)
-
+        dispatch(self, 'svm.SVC.fit', {
+            'onedal': self.__class__._onedal_fit,
+            'sklearn': sklearn_SVC.fit,
+        }, X, y, sample_weight)
         return self
 
+    @wrap_output_data
     def predict(self, X):
-        if hasattr(self, '_onedal_estimator'):
-            logging.info("sklearn.svm.SVC.predict: " + get_patch_message("onedal"))
-            return self._onedal_estimator.predict(X)
-        else:
-            logging.info("sklearn.svm.SVC.predict: " + get_patch_message("sklearn"))
-            return sklearn_SVC.predict(self, X)
+        return dispatch(self, 'svm.SVC.predict', {
+            'onedal': self.__class__._onedal_predict,
+            'sklearn': sklearn_SVC.predict,
+        }, X)
 
+    @wrap_output_data
     def _predict_proba(self, X):
-        if hasattr(self, '_onedal_estimator'):
-            logging.info("sklearn.svm.SVC._predict_proba: " + get_patch_message("onedal"))
-            if getattr(self, 'clf_prob', None) is None:
-                raise NotFittedError(
-                    "predict_proba is not available when fitted with probability=False")
-            return self.clf_prob.predict_proba(X)
-        else:
-            logging.info(
-                "sklearn.svm.SVC._predict_proba: " + get_patch_message("sklearn"))
-            return sklearn_SVC._predict_proba(self, X)
+        return dispatch(self, 'svm.SVC._predict_proba', {
+            'onedal': self.__class__._onedal_predict_proba,
+            'sklearn': sklearn_SVC._predict_proba,
+        }, X)
 
+    @wrap_output_data
     def decision_function(self, X):
-        if hasattr(self, '_onedal_estimator'):
-            logging.info(
-                "sklearn.svm.SVC.decision_function: " + get_patch_message("onedal"))
-            return self._onedal_estimator.decision_function(X)
-        else:
-            logging.info(
-                "sklearn.svm.SVC.decision_function: " + get_patch_message("sklearn"))
-            return sklearn_SVC.decision_function(self, X)
+        return dispatch(self, 'svm.SVC.decision_function', {
+            'onedal': self.__class__._onedal_decision_function,
+            'sklearn': sklearn_SVC.decision_function,
+        }, X)
+
+    def _onedal_gpu_supported(self, method_name, *data):
+        if method_name == 'svm.SVC.fit':
+            if len(data) > 1:
```

Comment:
else? or is this condition necessary at all?

Reply:
This condition is needed to initialize `_class_count` when fit support is checked. This function is also used in predict, decision_function, etc., where `_class_count` is already initialized by prior fit call. 
I can add `else` branch with exception throwing.